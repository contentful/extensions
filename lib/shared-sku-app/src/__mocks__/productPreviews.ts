import { Product } from '../interfaces';

const productPreviews: Product[] = [
  {
    id: '28a7442d',
    image: 'https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-1.jpg',
    name: 'Dress Twin-Set rose',
    sku: 'M0E20130820E90Z',
    externalLink: 'https://mc.commercetools.com/example-project/products/28a7442d/general'
  },
  {
    id: '4e5446e8',
    image: 'https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-2.jpg',
    name: 'Michael Kors – Shopper “Jet Set Travel”',
    sku: 'A0E2300FX102203',
    externalLink: 'https://mc.commercetools.com/example-project/products/4e5446e8/general'
  },
  {
    id: 'db3c234l',
    image: 'https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-3.jpg',
    name: 'Shirt “Jenny“ Polo Ralph Lauren white',
    sku: 'M0E21300900DZN7',
    externalLink: 'https://mc.commercetools.com/example-project/products/db3c234l/general'
  }
];

export default productPreviews;
