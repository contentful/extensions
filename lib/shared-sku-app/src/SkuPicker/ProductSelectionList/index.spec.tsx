import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Props, ProductSelectionList } from '.';
import productPreviews from '../../__mocks__/productPreviews';

const defaultProps: Props = {
  products: productPreviews,
  selectProduct: jest.fn()
};

const renderComponent = (props: Props) => {
  return render(<ProductSelectionList {...props} />);
};

describe('ProductSelectionList', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const component = renderComponent(defaultProps);
    expect(component.container).toMatchSnapshot();
  });
});
