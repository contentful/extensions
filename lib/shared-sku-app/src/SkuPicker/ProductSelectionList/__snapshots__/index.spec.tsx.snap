// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSelectionList should render successfully 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <div
      class="css-1uq6b1f"
    >
      <div
        aria-checked="true"
        class="css-4gm53r"
        data-test-id="selection-preview-M0E20130820E90Z"
        role="switch"
        tabindex="-1"
      >
        <span
          class="Tooltip__Tooltip__target-wrapper___Mtw42"
        >
          <div
            class="css-1v0ivop"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--white___3GVPJ"
              data-test-id="cf-ui-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <img
            alt="product preview"
            class="css-1ilamao"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-1.jpg"
          />
        </span>
      </div>
    </div>
    <div
      class="css-1uq6b1f"
    >
      <div
        aria-checked="true"
        class="css-4gm53r"
        data-test-id="selection-preview-A0E2300FX102203"
        role="switch"
        tabindex="-1"
      >
        <span
          class="Tooltip__Tooltip__target-wrapper___Mtw42"
        >
          <div
            class="css-1v0ivop"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--white___3GVPJ"
              data-test-id="cf-ui-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <img
            alt="product preview"
            class="css-1ilamao"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-2.jpg"
          />
        </span>
      </div>
    </div>
    <div
      class="css-1uq6b1f"
    >
      <div
        aria-checked="true"
        class="css-4gm53r"
        data-test-id="selection-preview-M0E21300900DZN7"
        role="switch"
        tabindex="-1"
      >
        <span
          class="Tooltip__Tooltip__target-wrapper___Mtw42"
        >
          <div
            class="css-1v0ivop"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--white___3GVPJ"
              data-test-id="cf-ui-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <img
            alt="product preview"
            class="css-1ilamao"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-3.jpg"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
